tracer/test/Datadog.Trace.TestHelpers/ProcessHelper.cs

```diff
@@ -59,17 +58,22 @@ public virtual void Dispose()
             }
         }
 
-        private void DrainOutput(DataReceivedEventArgs e, StringBuilder buffer, TaskCompletionSource<bool> tcs, Action<string> onDataReceived)
+        private void DrainOutput(StreamReader stream, StringBuilder buffer, TaskCompletionSource<bool> tcs, Action<string> onDataReceived)
         {
-            if (e.Data == null)
+            while (stream.ReadLine() is { } line)
```

Comment:
Two questions:

- Does this handle the case where the process exists and is disposed?
- This loops until it reaches the end of the stream, but that doesn't mean the process has exited right? There could be more data later?

Reply:
As long as the stream is open, `stream.ReadLine()` will block until a full line is read. It will only return null when the underlying stream is closed. 
I'm assuming that the stream will remain open until the process exits. If that assumption is correct then it should handle the case when the process exits (so there could be a race condition if we dispose it too quickly?), and the loop won't end until the process exits.